apps/meteor/server/modules/watchers/lib/messages.ts

```diff
@@ -12,41 +13,48 @@ const getUserNameCached = mem(
 	{ maxAge: 10000 },
 );
 
-export const broadcastMessageSentEvent = async ({
-	id,
-	data,
-	broadcastCallback,
-}: {
-	id: IMessage['_id'];
-	broadcastCallback: (message: IMessage) => Promise<void>;
-	data?: IMessage;
-}): Promise<void> => {
+export async function getMessageToBroadcast({ id, data }: { id: IMessage['_id']; data?: IMessage }): Promise<IMessage | void> {
 	const message = data ?? (await Messages.findOneById(id));
 	if (!message) {
 		return;
 	}
 
-	if (message._hidden !== true && message.imported == null) {
-		const UseRealName = (await getSettingCached('UI_Use_Real_Name')) === true;
+	if (message._hidden || message.imported != null) {
+		return;
+	}
 
-		if (UseRealName) {
-			if (message.u?._id) {
-				const name = await getUserNameCached(message.u._id);
-				if (name) {
-					message.u.name = name;
-				}
+	const UseRealName = (await getSettingCached('UI_Use_Real_Name')) === true;
+	if (UseRealName) {
+		if (message.u?._id) {
+			const name = await getUserNameCached(message.u._id);
+			if (name) {
+				message.u.name = name;
 			}
+		}
 
-			if (message.mentions?.length) {
-				for await (const mention of message.mentions) {
-					const name = await getUserNameCached(mention._id);
-					if (name) {
-						mention.name = name;
-					}
+		if (message.mentions?.length) {
+			for await (const mention of message.mentions) {
+				const name = await getUserNameCached(mention._id);
+				if (name) {
+					mention.name = name;
 				}
 			}
 		}
+	}
+
+	return message;
+}
 
-		void broadcastCallback(message);
+// TODO once the broadcast from file apps/meteor/server/modules/watchers/watchers.module.ts is removed
+// this function can be renamed to broadcastMessage
+export async function broadcastMessageFromData({ id, data }: { id: IMessage['_id']; data?: IMessage }): Promise<void> {
+	console.log('broadcastMessageFromData', dbWatchersDisabled);
+	if (!dbWatchersDisabled) {
+		return;
+	}
```

Comment:
```suggestion
	// if db watchers are active, the event will be triggered automatically so we don't need to broadcast it here.
	if (!dbWatchersDisabled) {
		return;
	}
```

Adding a comment here to just to clarify the difference